### ISSUE TITLE
pgBackRest failing fo read/validate files written to Swift S3

### ISSUE THREAD

**Issue by mtkunkel** [pgbackrest/pgbackrest#578]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: **pgBackRest 2.04**

2. PostgreSQL version:  **10.5**

3. Operating system/version - **Centos7**

4. Did you install pgBackRest from source or from a package? **PGDG  Package**

5. Please attach the following as applicable:

Will send others in email, unsure how strong some of those encrypted values are.

`archive_command = pgbackrest --stanza=concord-test_test01 archive-push %p`
`archive_mode = on`
`listen_address = '*'`
`max_wal_senders = 5`
` wal_level = replica`

7. Describe the issue:

pgBackRest is able to `stanza-create` and files are being written during backup and archive-push.  Both of these actions result in failure because pgbackrest is unable to properly read the files in S3 (Swift).

```
-bash-4.2$ pgbackrest --stanza=concord-test_test01 --log-level-console=trace check > pgbackrest_check.trace
ERROR [082]: : could not find WAL segment 000000010000000000000040 after 60 second(s)
ERROR [082]: : could not find WAL segment 000000010000000000000040 after 60 second(s)
WARN: WAL segment 000000010000000000000040 did not reach the archive:10-1
HINT: Check the archive_command to ensure that all options are correct (especially --stanza).
HINT: Check the PostgreSQL server log for errors.
```

OR During Backup

```
-bash-4.2$ pgbackrest --stanza=concord-test_test01 --log-level-console=info backup
2018-08-16 19:51:44.946 P00   INFO: backup command begin 2.04: --backup-standby --log-level-console=info --pg1-host=10.227.221.79 --pg1-path=/data/pg10/pgdata --process-max=1 --repo1-path=/concord-test_test01 --repo1-retention-diff=2 --repo1-retention-full=2 --repo1-s3-bucket=concord-test_test01 --repo1-s3-ca-file=/etc/pki/tls/certs/ca-bundle.crt --repo1-s3-endpoint=redacted.hostname --repo1-s3-host=redacted.hostname --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=dfw_obj_100 --repo1-type=s3 --stanza=concord-test_test01 --stop-auto
WARN: option backup-standby is enabled but standby is not properly configured - backups will be performed from the master
WARN: no prior backup exists, incr backup has been changed to full
2018-08-16 19:51:47.035 P00   INFO: execute non-exclusive pg_start_backup() with label "pgBackRest backup started at 2018-08-16 19:51:45": backup begins after the next regular checkpoint completes
2018-08-16 19:51:47.362 P00   INFO: backup start archive = 00000001000000000000003D, lsn = 0/3D000028
ERROR [055]: : raised from local-1 process: unable to get info for missing file /concord-test_test01/backup/concord-test_test01/20180816-195145F/pg_data/base/16388/16427.gz
```

#### Files are reaching storage: 
```
[mtkunke@lmtkunke5 oss]$ rclone lsjson DFW_S3_V4:concord-test_test01/backup/concord-test_test01/20180816-195145F/pg_data/base/16388/16427.gz
[
{"Path":"16427.gz","Name":"16427.gz","Size":3315495,"MimeType":"binary/octet-stream","ModTime":"2018-08-16T19:51:52Z","IsDir":false}
]
```

**Comment by cmwshang** [comment]
We'll take a look at this in depth when we receive the log and config files. In the meantime, please look at the following issues to see if there is something mentioned that might assist you.
https://github.com/pgbackrest/pgbackrest/issues/409 and https://github.com/pgbackrest/pgbackrest/issues/402

**Comment by dwsteele** [comment]
Well, this is certainly not good:
```
ERROR [055]: : raised from local-1 process: unable to get info for missing file /concord-test_test01/backup/concord-test_test01/20180816-195145F/pg_data/base/16388/16427.gz
```
My initial guess is that Swift has different guarantees for consistency than S3 does -- or it is configurable.  I know that pgBackRest is being used successfully on Swift so I wonder if this is configuration.  If Swift allows eventual consistency then we'll need to add support for that.  We plan to have a swift-native driver but for now we want to make sure the S3 gateway works.

I'm a bit baffled that there is no error in the postgres logs.  I'd be surprised if the file took more than 60 seconds to become visible, but I guess it's possible.  Can you try increasing `archive-timeout` to `300` and see what happens?  Any chance this cluster is behind on archiving?

**Comment by dwsteele** [comment]
So it appears that Swift can have stale lists even though the object is immediately available to be read.  Info is retrieved through a list command so that probably explains what is going on here.

Not sure why we haven't seen this before but you may have a larger/busier cluster than other users.

Seems pretty easy to add a wait in here, so that's probably what we'll do.

**Comment by dwsteele** [comment]
This has been fixed at 1ad67644dade53595e.  Sorry it took so long.

I had thought this issue affected Swift only which is why it did not get a high priority.  After doing some research it seems that S3 is also susceptible to this issue though it has never been reported and I have never seen it myself.

In any case it occurred to me that the size check was superfluous for S3 and other object stores so I just removed it and get the size from the copy operation instead.  Size checks are still required in resume but by that time (after a failed backup) the list should definitely be updated.
